diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 1e977e8..a6fadb8 100755
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -60,10 +60,20 @@ else()
         -Wmissing-declarations
         -Wno-attributes
         -Wno-unused-parameter
+        -Ofast
+        -fuse-linker-plugin
+        -flto
+        -funroll-loops
+        -faggressive-loop-optimizations
+        -floop-interchange
+        -floop-strip-mine
+        -floop-block
+        -fgraphite-identity
+        -ftree-loop-distribution
     )
 
     if (ARCHITECTURE_x86_64)
-        add_compile_options("-mcx16")
+        add_compile_options("-march=native")
     endif()
 
     if (APPLE AND CMAKE_CXX_COMPILER_ID STREQUAL Clang)
